src/main/scala/miss/supervisor/Supervisor.scala

```diff
@@ -22,117 +25,179 @@ class Supervisor(config: Config) extends Actor {
   private val verticalRoadsNum = cols * size
   private val horizontalRoadsNum = rows * size
 
-  private val actors = ofDim[ActorRef](rows, cols)
-  private val horizontalBoundaryActors = ofDim[ActorRef](rows)
-  private val verticalBoundaryActors = ofDim[ActorRef](cols)
-  private val horizontalRoadDefs = ofDim[RoadDefinition](horizontalRoadsNum)
-  private val verticalRoadDefs = ofDim[RoadDefinition](verticalRoadsNum)
+  private val workerNodesCount = config.getInt("worker.nodes")
+  private val workerCores = config.getInt("worker.cores")
 
-  @volatile var cityVisualizer: Option[ActorRef] = None
+  startWith(Initial, EmptyData)
 
-  override def receive: Receive = {
-    case Start =>
-      // Create area actors
-      for (i <- 0 until rows) {
-        for (j <- 0 until cols) {
-          actors(i)(j) = context.actorOf(Props(classOf[AreaActor]), s"AreaActor_${i}_$j")
-        }
+  when(Initial) {
+    case Event(Start, EmptyData) =>
+      log.warning("Waiting for workers")
+      goto(WaitingForWorkers) using WorkersData(List[ActorRef]())
+  }
+
+  when(WaitingForWorkers) {
+    case Event(RegisterWorker, w@WorkersData(workers)) =>
+      log.warning(sender().path.address.toString)
```

Comment:
This can cause race condition. Please do this:
```scala
val senderActor = sender()
log.warning(senderActor.path.address.toString)
val newWorkers = senderActor :: workers
```